juju/bundle.py

```diff
@@ -152,14 +156,63 @@ async def _handle_local_charms(self, bundle, bundle_dir):
                     app_name,
                     charm_url,
                     utils.get_local_charm_metadata(charm_dir),
-                    resources=bundle["applications"][app_name].get("resources", {}),
+                    resources=bundle.get('applications', bundle.get('services', {}))[app_name].get("resources", {}),
```

Comment:
Services are dead to us, we shouldn't introduce deprecated functionality.

Reply:
I'm pretty sure I added that because a bundle had 'services' but not 'applications', but maybe it was because of the facade versions etc.

Reply:
https://github.com/juju/charm/pull/326

Reply:
Yeah turns out I copy/pasted some old bundles to make new test bundles and introduced them myself, I updated all of them, running local tests right now, gonna push the patch soon :+1: